crates/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -285,6 +324,59 @@ fn vm_move_call<E: fmt::Debug, S: StorageView<E>>(
     Ok(result)
 }
 
+/// - Deserializes the modules
+/// - Publishes them into the VM, which invokes the Move verifier
+/// - Run the Sui Verifier
+fn publish_and_verify_modules<E: fmt::Debug, S: StorageView<E>>(
+    context: &mut ExecutionContext<E, S>,
+    module_bytes: Vec<Vec<u8>>,
+) -> Result<Vec<CompiledModule>, ExecutionError> {
+    let mut modules = module_bytes
+        .iter()
+        .map(|b| {
+            CompiledModule::deserialize(b)
+                .map_err(|e| e.finish(move_binary_format::errors::Location::Undefined))
+        })
+        .collect::<move_binary_format::errors::VMResult<Vec<CompiledModule>>>()
+        .map_err(|e| context.convert_vm_error(e))?;
+
+    if modules.is_empty() {
+        return Err(ExecutionErrorKind::PublishErrorEmptyPackage.into());
+    }
+
+    // It should be fine that this does not go through context.fresh_id since the Move runtime
+    // does not to know about new packages created, since Move objects and Move packages cannot
+    // interact
```

Comment:
I'm not sure I understand this comment -- `generate_package_id` calls `ctx.fresh_id`:

https://github.com/MystenLabs/sui/blob/6ff0c785f5ab7f4857531b117347bdf959f6651c/crates/sui-adapter/src/adapter.rs#L575-L579

Reply:
wrong context, I'll clarify, its for `ExecutionContext::fresh_id` since that one needs to talk to the Move runtime too